reporter/errors.go

```diff
@@ -66,3 +66,49 @@ func (e errorWithSourcePos) Unwrap() error {
 }
 
 var _ ErrorWithPos = errorWithSourcePos{}
+
+type ErrorWithNode interface {
+	ErrorWithPos
```

Comment:
We could additionally provide an GetEndPosition() function on ErrorWithPos that return the position of node.End() when available.

Reply:
I was wondering if it might be better to call this `ErrorWithSpan` and to just add `GetEndPosition()`. Do we really need to expose the actual node and file? The actual implementation could just store both positions, instead of the node and file. That way this error interface isn't so tightly coupled to the AST, which would make it easier to implement. For example, it could be implemented without an AST, using only details from a `SourceCodeInfo` in the file descriptor.